README.rst

```diff
@@ -277,47 +294,29 @@ But this poses a few disadvantages:
   naming fixtures as parameters, or ``pytest.mark.parametrize``;
 - you can't easily undo the mocking during the test execution;
 
-
-**Note about usage as context manager**
-
-Although mocker's API is intentionally the same as ``mock.patch``'s, its use
-as context manager and function decorator is **not** supported through the
-fixture. The purpose of this plugin is to make the use of context managers and
-function decorators for mocking unnecessary. Indeed, trying to use the
-functionality in ``mocker`` in this manner can lead to non-intuitive errors:
+An alternative is to use ``contextlib.ExitStack`` to stack the context managers in a single level of indentation 
+to improve the flow of the test:
 
 .. code-block:: python
 
-    def test_context_manager(mocker):
-        a = A()
-        with mocker.patch.object(a, 'doIt', return_value=True, autospec=True):
-            assert a.doIt() == True
-
-.. code-block:: console
-
-    ================================== FAILURES ===================================
-    ____________________________ test_context_manager _____________________________
-    in test_context_manager
-        with mocker.patch.object(a, 'doIt', return_value=True, autospec=True):
-    E   AttributeError: __exit__
-
-You can however use ``mocker.mock_module`` to access the underlying ``mock``
-module, e.g. to return a context manager in a fixture that mocks something
-temporarily:
+    import contextlib
+    import mock
 
-.. code-block:: python
+    def test_unix_fs():
+        with contextlib.ExitStack() as stack:
+            stack.enter_context(mock.patch('os.remove'))
+            UnixFS.rm('file')
+            os.remove.assert_called_once_with('file')
 
-    @pytest.fixture
-    def fixture_cm(mocker):
-        @contextlib.contextmanager
-        def my_cm():
-            def mocked():
-                pass
+            stack.enter_context(mock.patch('os.listdir'))
+            assert UnixFS.ls('dir') == expected
+            # ...
 
-            with mocker.mock_module.patch.object(SomeClass, 'method', mocked):
-                yield
-        return my_cm
+            stack.enter_context(mock.patch('shutil.copy'))
+            UnixFS.cp('src', 'dst')
+            # ...
 
+But this is arguably a little more complex.
```

Comment:
This is a tricky statement to justify, and it's not clear what is more complex that what.

dym, "Using ``contextlib.ExitStack`` is arguably a little more complex than"...
"pytest-mock" or "using with statements" or "using mock.patch decorators"

Reply:
it's tricky to justify because you could also argue that ``contextlib.ExitStack`` is the simplest of the approaches listed.

Reply:
Agreed, done. 👍